dateutil/parser.py

```diff
@@ -1328,9 +1327,6 @@ def _parse_hms(i, l, info, res):
     return i
 
 
-# TODO: require len(token) >= 3 like we do for the between-parens version?
```

Comment:
Yeah, the time zone stuff needs a pretty serious overhaul. The reason I haven't really started into it totally yet is that I think it involves some opportunistic loading of time zone data from various sources, and I'm not sure how heavy a burden that will be in resource-constrained environments, so I've been trying to think through the right way to do it.

I'm not sure why the between-parens tzname criteria is different, but one thing I've learned is that a lot of the stuff that just seems crazy in the parser is there because there's some spec or some subset of people generating date strings in a very specific way (like the fact that `12:00:00+0300` and '12:00:00 UTC+0300` parse to UTC + 3 and UTC **-3**, respectively), so it's probably worth trying to dig up why they were separate in the first place before unifying them.

That said, we should definitely maximize code-reuse if only so that downstream users only have to patch in a single place if they want a different behavior, so if there's some reason they are different, it's worth factoring out everything except that difference.